shopfloor/actions/data.py

```diff
@@ -29,13 +29,20 @@ def _location_parser(self):
 
     @ensure_model("stock.picking")
     def picking(self, record, **kw):
-        return self._jsonify(record, self._picking_parser, **kw)
+        parser = self._picking_parser
+        # progress is a heavy computed field,
+        # and it may reduce performance significatively
+        # when dealing with a large number of pickings.
+        # Thus, we make it optional.
```

Comment:
It is not a heavy computed field.
It is a stored field, it is not computed on demand.
So, retrieving it is pretty light, and it wouldn't hurt to have it, but I'll let the others decide.

Reply:
I agree.

@simahawk thoughts?

Reply:
Ok with @mmequignon

Reply:
the 1st time you compute this it will require time. I'd keep it optional.

Reply:
There's a pre-init hook in stock_picking_progress, shouldn't be an issue. But it's ok.